### PR TITLE
Add `ECSTask.kill` for cancellation support

### PR DESCRIPTION
**Pull request by madkinsz**
Part of https://github.com/PrefectHQ/prefect/pull/7666

### Checklist

- ~[ ] This pull request references any related issue by including "Closes #<ISSUE_NUMBER>"~
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-aws/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-aws/blob/main/CHANGELOG.md)

### CODE REVIEW

File: tests/test_ecs.py [hunk 1]
```diff
@@ -1508,3 +1505,140 @@ async def test_family_from_task_definition_arn(aws_credentials, prepare_for_flow
     task = describe_task(ecs_client, task_arn)
     task_definition = describe_task_definition(ecs_client, task)
     assert task_definition["family"] == "test-family"
+
+
+@pytest.mark.usefixtures("ecs_mocks")
+@pytest.mark.usefixtures("ecs_mocks")
```
Comment (madkinsz) [inline tests/test_ecs.py:1511]:
```suggestion
@pytest.mark.usefixtures("ecs_mocks")
```

Comment (madkinsz) [reply tests/test_ecs.py:1511]: Duplicate

File: prefect_aws/ecs.py [hunk 2]
```diff
@@ -531,12 +543,63 @@ async def run(self, task_status: Optional[TaskStatus] = None) -> ECSTaskResult:
...
+        Args:
+            identifier: A cluster and task arn combination. This should match a value
+                yielded by `ECSTask.run`.
+        """
+        if grace_period != 30:
+            self.logger.warning(
+                "Kill grace period of {grace_period}s requested, but AWS does not "
+                "support dynamic grace period configuration. "
+                "See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-config.html for configuration of grace periods."  # noqa
+            )
```
Comment (ahuang11) [inline prefect_aws/ecs.py:566]:
```suggestion
            self.logger.warning(
                "Kill grace period of %ss requested, but AWS does not "
                "support dynamic grace period configuration. "
                "See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-config.html for configuration of grace periods.",  # noqa
                grace_period
            )
```

Comment (madkinsz) [reply prefect_aws/ecs.py:566]: We do not use that (ik, more correct) style anywhere in Prefect and notably nowhere in this file. I think we should stick to the style of the current file, at least. We can change these to be the proper "delayed" formats later. 

This did make me notice that there's not a test for this though!

Comment (ahuang11) [reply prefect_aws/ecs.py:566]: Okay; that's fine with me. In that case, I think the string needs to be converted to f-string if I'm not mistaken.

Comment (madkinsz) [reply prefect_aws/ecs.py:566]: 👍 yeah good catch.

**Comment by madkinsz** [comment]
Requires https://github.com/PrefectHQ/prefect/pull/7666

@anna-geller this should be testable with an agent on that branch if you're interested in doing the QA!

**Review by anna-geller — APPROVED** [review]
Works 🎉 
image

image

image

File: prefect_aws/ecs.py [hunk 3]
```diff
@@ -160,6 +161,11 @@ def get_container(containers: List[dict], name: str) -> Optional[dict]:
     return None
 
 
+def parse_task_identifier(identifier: str) -> Tuple[str, str]:
+    cluster, task = identifier.split("::", maxsplit=1)
```
Comment (ahuang11) [inline prefect_aws/ecs.py:165]:
```suggestion
def parse_task_identifier(identifier: str) -> Tuple[str, str]:
    """
    Splits identifier into its cluster and task components, e.g.
    input "cluster_name::task_arn" outputs ("cluster_name", "task_arn").
    """
    cluster, task = identifier.split("::", maxsplit=1)
```

File: prefect_aws/ecs.py [hunk 4]
```diff
@@ -531,12 +543,63 @@ async def run(self, task_status: Optional[TaskStatus] = None) -> ECSTaskResult:
...
+                ) from exc
+            if "not find task" in str(exc):
+                raise InfrastructureNotFound(
+                    f"Cannot stop ECS task: the task {task!r} could not be found in "
+                    f"cluster {cluster!r}."
+                ) from exc
+            if "no registered tasks" in str(exc):
+                raise InfrastructureNotFound(
+                    f"Cannot stop ECS task: the cluster {cluster!r} has no tasks."
+                ) from exc
```
Comment (ahuang11) [inline prefect_aws/ecs.py:598]:
```suggestion
            exc_msg = str(exc)
            if "ClusterNotFound" in exc_msg:
                raise InfrastructureNotFound(
                    f"Cannot stop ECS task: the cluster {cluster!r} could not be found."
                ) from exc
            if "not find task" in exc_msg:
                raise InfrastructureNotFound(
                    f"Cannot stop ECS task: the task {task!r} could not be found in "
                    f"cluster {cluster!r}."
                ) from exc
            if "no registered tasks" in exc_msg:
                raise InfrastructureNotFound(
                    f"Cannot stop ECS task: the cluster {cluster!r} has no tasks."
                ) from exc
```

**Review by ahuang11 — APPROVED** [review]
A couple suggestions, but looks good to me!

Note to whoever releases this, the next version should be `0.2.0` due to breaking change.

File: prefect_aws/ecs.py (hunk 4 above)
Comment (madkinsz) [reply prefect_aws/ecs.py:602]: I'm not sure this is a particularly meaningful change for performance, is this for readability?

Comment (ahuang11) [reply prefect_aws/ecs.py:602]: I personally like to define a variable if used multiple times, but I don't feel strongly about it.

**Pull request merged**